File: data-pipeline/redis-updater/src/test/java/org/ekstep/ep/samza/service/Fixtures/EventFixture.java
package org.ekstep.ep.samza.service.Fixtures;

public class EventFixture {
    public static final String OBJECT_TYPE_CONTENT_EVENT_1 = "{\"ets\":1548780816826,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"ownershipType\":{\"ov\":null,\"nv\":[\"createdBy\"]},\"code\":{\"ov\":null,\"nv\":\"testbook1\"},\"channel\":{\"ov\":null,\"nv\":\"in.ekstep\"},\"description\":{\"ov\":null,\"nv\":\"TestCollection\"},\"status\":{\"ov\":null,\"nv\":\"Draft\"}}},\"label\":\"TestBook1\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-29T22:23:36.826+0530\",\"objectType\":\"Content\",\"nodeUniqueId\":\"do_11268761245100441611\",\"requestId\":null,\"operationType\":\"CREATE\",\"nodeGraphId\":703654,\"graphId\":\"domain\"}";
    public static final String OBJECT_TYPE_CONTENT_EVENT_1_UPDATED = "{\"ets\":1548780816826,\"channel\":\"sunbird.portal\",\"transactionData\":{\"properties\":{\"ownershipType\":{\"ov\":null,\"nv\":null},\"code\":{\"ov\":null,\"nv\":\"testbook1\"},\"channel\":{\"ov\":null,\"nv\":\"sunbird.portal\"},\"status\":{\"ov\":null,\"nv\":\"Live\"},\"visibility\":{\"ov\":null,\"nv\":\"Default\"}}},\"label\":\"TestBook1\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-29T22:23:36.826+0530\",\"objectType\":\"Content\",\"nodeUniqueId\":\"do_11268761245100441611\",\"requestId\":null,\"operationType\":\"CREATE\",\"nodeGraphId\":703654,\"graphId\":\"domain\"}";
    public static final String OBJECT_TYPE_DIAL_CODE_1 = "{\"nodeUniqueId\":\"YC9EP8\",\"ets\":1549345256766,\"requestId\":null,\"audit\":false,\"transactionData\":{\"properties\":{\"dialcode_index\":{\"ov\":null,\"nv\":261351.0},\"identifier\":{\"ov\":null,\"nv\":\"YC9EP8\"},\"channel\":{\"ov\":null,\"nv\":\"b00bc992ef25f1a9a8d63291e20efc8d\"},\"batchcode\":{\"ov\":null,\"nv\":\"do_112692236142379008136\"},\"publisher\":{\"ov\":null,\"nv\":null},\"generated_on\":{\"ov\":null,\"nv\":\"2019-02-05T05:40:56.762\"},\"status\":{\"ov\":null,\"nv\":\"Draft\"}}},\"channel\":\"in.ekstep\",\"index\":true,\"operationType\":\"CREATE\",\"nodeType\":\"EXTERNAL\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-02-05T05:40:56.766+0000\",\"objectType\":\"DialCode\"}";
    public static final String CONTENT_EVENT_EMPTY_PROPERTIES = "{\"ets\":1548780816826,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{}},\"label\":\"TestBook1\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-29T22:23:36.826+0530\",\"objectType\":\"Content\",\"nodeUniqueId\":\"do_11268761245100441611\",\"requestId\":null,\"operationType\":\"CREATE\",\"nodeGraphId\":703654,\"graphId\":\"domain\"}";
    public static final String CONTENT_EVENT_EMPTY_NODE_UNIQUEID = "{\"ets\":1548780816826,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"ownershipType\":{\"ov\":null,\"nv\":[\"createdBy\"]},\"code\":{\"ov\":null,\"nv\":\"testbook1\"},\"channel\":{\"ov\":null,\"nv\":\"in.ekstep\"},\"description\":{\"ov\":null,\"nv\":\"TestCollection\"},\"status\":{\"ov\":null,\"nv\":\"Draft\"}}},\"label\":\"TestBook1\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-29T22:23:36.826+0530\",\"objectType\":\"Content\",\"nodeUniqueId\":\"\",\"requestId\":null,\"operationType\":\"CREATE\",\"nodeGraphId\":703654,\"graphId\":\"domain\"}";
    public static final String OBJECT_TYPE_CONCEPT_EVENT = "{\"ets\":1547644874028,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"language\":{\"ov\":null,\"nv\":\"English\"}}},\"label\":\"Properties of Light\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-16T18:51:14.028+0530\",\"objectType\":\"Concept\",\"nodeUniqueId\":\"SMC4\",\"requestId\":null,\"operationType\":\"UPDATE\",\"nodeGraphId\":101409,\"graphId\":\"domain\"}";
    public static final String EMPTY_LANGUAGE_FIELD_EVENT= "{\"ets\":1547644874028,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"language\":{\"ov\":null,\"nv\":\"\"}}},\"label\":\"Properties of Light\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-16T18:51:14.028+0530\",\"objectType\":\"Concept\",\"nodeUniqueId\":\"SMC4\",\"requestId\":null,\"operationType\":\"UPDATE\",\"nodeGraphId\":101409,\"graphId\":\"domain\"}";
    public static final String LIST_LANGUAGE_FIELD_EVENT= "{\"ets\":1547644874028,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"language\":{\"ov\":null,\"nv\":[\"Kannada\"]}}},\"label\":\"Properties of Light\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-01-16T18:51:14.028+0530\",\"objectType\":\"Concept\",\"nodeUniqueId\":\"SMC4\",\"requestId\":null,\"operationType\":\"UPDATE\",\"nodeGraphId\":101409,\"graphId\":\"domain\"}";
    public static final String INCORRECT_LIST_TYPE_FIELDS= "{\"ets\":1559670759,\"channel\":\"in.ekstep\",\"transactionData\":{\"properties\":{\"language\":{\"ov\":null,\"nv\":\"Spanish\"},\"subject\":{\"ov\":null,\"nv\":\"CS\"},\"ageGroup\":{\"ov\":null,\"nv\":\"22\"},\"ownershipType\":{\"ov\":null,\"nv\":[\"createdBy\"]}}},\"label\":\"Properties of Light\",\"nodeType\":\"DATA_NODE\",\"userId\":\"ANONYMOUS\",\"createdOn\":\"2019-06-07T18:51:14.028+0530\",\"objectType\":\"Concept\",\"nodeUniqueId\":\"SMC4\",\"requestId\":null,\"operationType\":\"UPDATE\",\"nodeGraphId\":98765,\"graphId\":\"domain\"}";
    public static final String AUDIT_EVENT_SIGINTYPE="{\n" +
            "  \"actor\": {\n" +
            "    \"type\": \"Consumer\",\n" +
            "    \"id\": \"89490534-126f-4f0b-82ac-3ff3e49f3468\"\n" +
            "  },\n" +
            "  \"eid\": \"AUDIT\",\n" +
            "  \"edata\": {\n" +
            "    \"state\": \"Create\",\n" +
            "    \"props\": [\n" +
            "      \"firstName\",\n" +
            "      \"email\",\n" +
            "      \"emailVerified\",\n" +
            "      \"id\",\n" +
            "      \"userId\",\n" +
            "      \"createdBy\",\n" +
            "      \"rootOrgId\",\n" +
            "      \"channel\",\n" +
            "      \"userType\",\n" +
            "      \"roles\",\n" +
            "      \"phoneVerified\",\n" +
            "      \"isDeleted\",\n" +
            "      \"createdDate\",\n" +
            "      \"status\",\n" +
            "      \"userName\",\n" +
            "      \"loginId\",\n" +
            "      \"externalIds\"\n" +
            "    ]\n" +
            "  },\n" +
            "  \"ver\": \"3.0\",\n" +
            "  \"ets\": 1561739226844,\n" +
            "  \"context\": {\n" +
            "    \"channel\": \"0126684405014528002\",\n" +
            "    \"pdata\": {\n" +
            "      \"pid\": \"learner-service\",\n" +
            "      \"ver\": \"2.0.0\",\n" +
            "      \"id\": \"prod.diksha.learning.service\"\n" +
            "    },\n" +
            "    \"env\": \"User\",\n" +
            "    \"cdata\": [\n" +
            "      {\n" +
            "        \"type\": \"User\",\n" +
            "        \"id\": \"34881c3a-8b92-4a3c-a982-7f946137cb09\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"SignupType\",\n" +
            "        \"id\": \"google\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"Source\",\n" +
            "        \"id\": \"android\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"Request\",\n" +
            "        \"id\": \"91f3c280-99c1-11e9-956e-6b6ef71ed575\"\n" +
            "      }\n" +
            "    ],\n" +
            "    \"rollup\": {\n" +
            "      \"l1\": \"0126684405014528002\"\n" +
            "    }\n" +
            "  },\n" +
            "  \"mid\": \"1561739226844.e0048ef8-a01e-4780-8c83-e571f28c53c8\",\n" +
            "  \"object\": {\n" +
            "    \"type\": \"User\",\n" +
            "    \"id\": \"89490534-126f-4f0b-82ac-3ff3e49f3468\"\n" +
            "  },\n" +
            "  \"syncts\": 1561739243532,\n" +
            "  \"@timestamp\": \"2019-06-28T16:27:23.532Z\",\n" +
            "  \"flags\": {\n" +
            "    \"tv_processed\": true\n" +
            "  },\n" +
            "  \"type\": \"events\"\n" +
            "}";
    public static final String AUDIT_EVENT_LOGININTYPE ="{\n" +
            "  \"actor\": {\n" +
            "    \"type\": \"System\",\n" +
            "    \"id\": \"3b46b4c9-3a10-439a-a2cb-feb5435b3a0d\"\n" +
            "  },\n" +
            "  \"eid\": \"AUDIT\",\n" +
            "  \"edata\": {\n" +
            "    \"state\": \"Updated\",\n" +
            "    \"props\": [\n" +
            "      \"medium\",\n" +
            "      \"board\",\n" +
            "      \"grade\",\n" +
            "      \"syllabus\",\n" +
            "      \"gradeValue\"\n" +
            "    ]\n" +
            "  },\n" +
            "  \"ver\": \"3.0\",\n" +
            "  \"ets\": 1561739240727,\n" +
            "  \"context\": {\n" +
            "    \"pdata\": {\n" +
            "      \"pid\": \"sunbird.app\",\n" +
            "      \"ver\": \"2.1.92\",\n" +
            "      \"id\": \"prod.diksha.app\"\n" +
            "    },\n" +
            "    \"channel\": \"505c7c48ac6dc1edc9b08f21db5a571d\",\n" +
            "    \"env\": \"sdk\",\n" +
            "    \"did\": \"010612971a80a7677d0a3e849ab35cb4a83157de\",\n" +
            "    \"cdata\": [\n" +
            "      {\n" +
            "        \"type\": \"UserRole\",\n" +
            "        \"id\": \"student\"\n" +
            "      }\n" +
            "    ],\n" +
            "    \"sid\": \"ea68a05e-0843-4c06-9a84-9b98cd974724\"\n" +
            "  },\n" +
            "  \"mid\": \"0268860e-76b0-4b4e-b99b-ebf543e7a9d8\",\n" +
            "  \"object\": {\n" +
            "    \"id\": \"3b46b4c9-3a10-439a-a2cb-feb5435b3a0d\",\n" +
            "    \"type\": \"User\",\n" +
            "    \"version\": \"\",\n" +
            "    \"rollup\": {}\n" +
            "  },\n" +
            "  \"syncts\": 1561739245463,\n" +
            "  \"@timestamp\": \"2019-06-28T16:27:25.463Z\",\n" +
            "  \"flags\": {\n" +
            "    \"tv_processed\": true\n" +
            "  },\n" +
            "  \"type\": \"events\"\n" +
            "}";

    public static final String AUDIT_EVENT_METADATA_UPDATED = "{\n" +
            "  \"actor\": {\n" +
            "    \"type\": \"Consumer\",\n" +
            "    \"id\": \"52226956-61d8-4c1b-b115-c660111866d34\"\n" +
            "  },\n" +
            "  \"eid\": \"AUDIT\",\n" +
            "  \"edata\": {\n" +
            "    \"state\": \"Update\",\n" +
            "    \"props\": [\n" +
            "      \"firstName\",\n" +
            "      \"email\",\n" +
            "      \"locationids\",\n" +
            "      \"emailVerified\"\n" +
            "    ]\n" +
            "  },\n" +
            "  \"ver\": \"3.0\",\n" +
            "  \"ets\": 1561739226844,\n" +
            "  \"context\": {\n" +
            "    \"channel\": \"0126684405014528002\",\n" +
            "    \"pdata\": {\n" +
            "      \"pid\": \"learner-service\",\n" +
            "      \"ver\": \"2.0.0\",\n" +
            "      \"id\": \"prod.diksha.learning.service\"\n" +
            "    },\n" +
            "    \"env\": \"User\",\n" +
            "    \"cdata\": [\n" +
            "      {\n" +
            "        \"type\": \"User\",\n" +
            "        \"id\": \"34881c3a-8b92-4a3c-a982-7f946137cb09\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"SignupType\",\n" +
            "        \"id\": \"google\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"Source\",\n" +
            "        \"id\": \"android\"\n" +
            "      },\n" +
            "      {\n" +
            "        \"type\": \"Request\",\n" +
            "        \"id\": \"91f3c280-99c1-11e9-956e-6b6ef71ed575\"\n" +
            "      }\n" +
            "    ],\n" +
            "    \"rollup\": {\n" +
            "      \"l1\": \"0126684405014528002\"\n" +
            "    }\n" +
            "  },\n" +
            "  \"mid\": \"1561739226844.e0048ef8-a01e-4780-8c83-e571f28c53c8\",\n" +
            "  \"object\": {\n" +
            "    \"type\": \"User\",\n" +
            "    \"id\": \"52226956-61d8-4c1b-b115-c660111866d3\"\n" +
            "  },\n" +
            "  \"syncts\": 1561739243532,\n" +
            "  \"@timestamp\": \"2019-06-28T16:27:23.532Z\",\n" +
            "  \"flags\": {\n" +
            "    \"tv_processed\": true\n" +
            "  },\n" +
            "  \"type\": \"events\"\n" +
            "}";
}

File: data-pipeline-flink/user-cache-updater-2.0/src/test/scala/org/sunbird/dp/fixture/EventFixture.scala
package org.sunbird.dp.fixture

object EventFixture {

  val telemetrEvents: List[String] = List(

    /**
      * User-Id : user-1
      * Edata.state is update
      * Having all the user data
      */
    """
      {"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-1","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin,
    /**
      * UserId: user-9
      * Client_Error
      */
    """{"actor":{"id":"cb0defc7-9f5f-4c0a-bb73-207d34ed2cf0","type":"User"},"eid":"AUDIT","edata":{"state":"Update","type":"mergeUser","props":["fromAccountId","toAccountId","type"]},"ver":"3.0","syncts":1621593310099,"@timestamp":"2021-05-21T10:35:10.099Z","ets":1621593307552,"context":{"channel":"0126796199493140480","pdata":{"id":"preprod.diksha.learning.service","pid":"learner-service","ver":"3.9.0"},"env":"User","cdata":[{"id":"9e6f3392-af99-425b-9404-e8e4078a1b65","type":"FromAccountId"},{"id":"cb0defc7-9f5f-4c0a-bb73-207d34ed2cf0","type":"ToAccountId"},{"id":"8f0d93fd-33ac-4a64-a39c-f008011c976a","type":"Request"}],"rollup":{"l1":"01275678925675724817"}},"flags":{"pp_duplicate_skipped":true,"pp_validation_processed":true},"mid":"8f0d93fd-33ac-4a64-a39c-f008011c976a","type":"events","object":{"id":"user-8","type":"User"}}""",
    /**
      * User-Id : user-2
      * Edata.state is update
      * user's data missed from API
      */
    """
      {"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-2","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin,
    /**
      * User-Id : user-3
      * Edata.state is update
      * framework and block info not present
      */
    """
      {"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-3","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin,
    /**
      * User-Id : user-4
      * Edata.state is update
      * framework and block info not present
      */
    """
      {"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-4","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin,
    /**
      * UserId = user-5
      * EData state is "Created"
      * User SignupType is "sso"
      * It should able to insert The Map(usersignintype, Validated)
      *
      */
        """
          |{"actor":{"type":"Consumer","id":"89490534-126f-4f0b-82ac-3ff3e49f3468"},"eid":"AUDIT","edata":{"state":"Created","props":["firstName","email","emailVerified","id","userId","createdBy","rootOrgId","channel","userType","roles","phoneVerified","isDeleted","createdDate","status","userName","loginId","externalIds"]},"ver":"3.0","ets":1561739226844,"context":{"channel":"0126684405014528002","pdata":{"pid":"learner-service","ver":"2.0.0","id":"prod.diksha.learning.service"},"env":"User","cdata":[{"type":"User","id":"34881c3a-8b92-4a3c-a982-7f946137cb09"},{"type":"SignupType","id":"sso"},{"type":"Source","id":"android"},{"type":"Request","id":"91f3c280-99c1-11e9-956e-6b6ef71ed575"}],"rollup":{"l1":"0126684405014528002"}},"mid":"1561739226844.e0048ef8-a01e-4780-8c83-e571f28c53c8","object":{"type":"User","id":"user-5"},"syncts":1561739243532,"@timestamp":"2019-06-28T16:27:23.532Z","flags":{"tv_processed":true},"type":"events"}""".stripMargin,

    /**
      * AUDIT Event with object type as content. (not related to user)
      * Props are not defined - Skip Count should incr (skipCount =+ 1)
      */
    """
      |{"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-5","cdata":[{"id":"25cb0530-7c52-ecb1-cff2-6a14faab7910","type":"UserRole"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"do_r8r8rew97we9r8","type":"content"},"edata":{"state":"Updated"},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin,

    /**
      * AUDIT Event, But having invalid state(other than create/update)
      * userid= user-6
      */

    """
      |
      |{"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-5","cdata":[{"id":"25cb0530-7c52-ecb1-cff2-6a14faab7910","type":"UserRole"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-7","type":"user"},"edata":{"state":"wrongState"},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |""".stripMargin,
    /**
     * User-Id : user-10
     * schoolname starting with '['
     * Having all the user data
     */
    """
      {"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-10","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |
      |""".stripMargin

  )

  /**
   * User-Id : user-1
   * Edata.state is update
   * Having pid as sunbird.app
   */
  val telemetryEventWithAppPid: String = """
      |{"actor":{"type":"System","id":"cb0defc7-9f5f-4c0a-bb73-207d34ed2cf0"},"eid":"AUDIT","edata":{"state":"Updated","props":[]},"ver":"3.0","syncts":1631551986826,"@timestamp":"2021-09-13T16:53:06.826Z","ets":1.63155195956E12,"context":{"cdata":[{"id":"teacher","type":"UserRole"},{"type":"Tabs","id":"Library-Course"},{"id":"a9d1d5df-da2e-4006-8495-363cce1ceec2","type":"UserSession"}],"env":"sdk","channel":"0126684405014528002","pdata":{"id":"dev.sunbird.app","pid":"sunbird.app","ver":"4.1.892"},"sid":"a9d1d5df-da2e-4006-8495-363cce1ceec2","did":"575fc6d8a853d2f543e4ef54d57aa3b6cd0110cf","rollup":{"l1":"0126684405014528002"}},"flags":{"ex_processed":true,"pp_validation_processed":true,"pp_duplicate_skipped":true},"mid":"bb429169-f320-4da3-80ad-660093aafe3b","type":"events","object":{"id":"user-1","type":"user","version":"","rollup":{}}}
      |""".stripMargin

  /**
   * User-Id : user-1
   * Edata.state is update
   * Having pid as learner-service
   */
  val telemetryEventWithLearnerPid: String = """
      |{"eid":"AUDIT","ets":1573121861118,"ver":"3.0","mid":"1573121861118.40f9136b-1cc3-458d-a04a-4459606df","actor":{"id":"5609876543234567890987654345678","type":"Request"},"context":{"channel":"01285019302823526477","pdata":{"id":"dev.sunbird.portal","pid":"learner-service","ver":"2.5.0"},"env":"User","did":"user-3","cdata":[{"id":"google","type":"SignupType"}],"rollup":{"l1":"01285019302823526477"}},"object":{"id":"user-1","type":"user"},"edata":{"state":"Update","props":["recoveryEmail","recoveryPhone","userId","id","externalIds","updatedDate","updatedBy"]},"syncts":1573121861125,"@timestamp":"2019-11-07T10:17:41.125Z","flags":{"tv_processed":true,"dd_processed":true},"type":"events","ts":"2019-11-07T10:17:41.118+0000"}
      |""".stripMargin

}